src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -221,42 +268,51 @@ public async Task StartContainerAsync(IExecutionContext executionContext, object
+                        node = container.ContainerBringNodePath;
                     }
-
-                    // Expose docker network to env
-                    executionContext.Variables.Set(Constants.Variables.Agent.ContainerNetwork, container.ContainerNetwork);
+                    else
+                    {
+                        node = container.TranslateToContainerPath(Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Externals), "node", "bin", $"node{IOUtil.ExeExtension}"));
+                    }
+                    string sleepCommand = $"\"{node}\" -e \"setInterval(function(){{}}, 24 * 60 * 60 * 1000);\"";
+                    container.ContainerCommand = sleepCommand;
                 }
-#endif
-                // See if this container brings its own Node.js
-                container.ContainerBringNodePath = await _dockerManger.DockerInspect(context: executionContext,
-                                                                      dockerObject: container.ContainerImage,
-                                                                      options: $"--format=\"{{{{index .Config.Labels \\\"{_nodeJsPathLabel}\\\"}}}}\"");
-
-                container.ContainerId = await _dockerManger.DockerCreate(context: executionContext,
-                                                                         displayName: container.ContainerDisplayName,
-                                                                         image: container.ContainerImage,
-                                                                         mountVolumes: container.MountVolumes,
-                                                                         network: container.ContainerNetwork,
-                                                                         options: container.ContainerCreateOptions,
-                                                                         environment: container.ContainerEnvironmentVariables);
+
+                container.ContainerId = await _dockerManger.DockerCreate(executionContext, container);
                 ArgUtil.NotNullOrEmpty(container.ContainerId, nameof(container.ContainerId));
-                executionContext.Variables.Set(Constants.Variables.Agent.ContainerId, container.ContainerId);
+                if (container.IsJobContainer)
+                {
+                    executionContext.Variables.Set(Constants.Variables.Agent.ContainerId, container.ContainerId);
+                }
 
                 // Start container
                 int startExitCode = await _dockerManger.DockerStart(executionContext, container.ContainerId);
                 if (startExitCode != 0)
                 {
                     throw new InvalidOperationException($"Docker start fail with exit code {startExitCode}");
                 }
+
+                // Get port mappings of running container
```

Comment:
i think you want to do this after we make sure the container is up and running?

Reply:
I agree. Fixed